discord/errors.py

```diff
@@ -125,6 +125,13 @@ class NotFound(HTTPException):
     pass
 
 
+class UnknownType(ClientException):
```

Comment:
I'm not sure I like this exception nor its docstring. I would prefer something like:

```py
class InvalidData(ClientException):
    """Exception that's raised when the library encounters unknown or invalid data from Discord."""
    pass
```

Easier and more future proof.